src/main/java/com/thegrizzlylabs/sardineandroid/Sardine.java

```diff
@@ -197,7 +200,7 @@
 	// InputStream not supported
 	// See https://github.com/square/okhttp/issues/2424
 
-	void put(String url, InputStream dataStream) throws IOException;
+	void put(ContentResolver cr, String url, Uri dataStream) throws IOException;
```

Comment:
I would keep the `url` first as to match other requests and to avoid confusing it with the `Uri`